resources/nomad_client.go

```diff
@@ -84,16 +86,47 @@ func (nc NomadClient) Scale(newCount int, vc *VaultClient) error {
 	*tg.Count = newCount
 	*job.VaultToken = vc.GetVaultToken()
 
-	logging.Info("Old nomad: %d. Desired nomad: %d", oldCount, newCount)
-
 	_, _, err = nc.client.nomad.Jobs().Register(job, &nomad.WriteOptions{})
 	if err != nil {
 		logging.Error(err.Error())
 		return err
 	}
 
 	count, _ := nc.GetTaskGroupCount()
-	logging.Info("Old nomad: %d. New nomad: %d", oldCount, count)
+	logging.Info("[scaling log] Nomad job: %s. Old: %d. New: %d", nc.JobName, oldCount, count)
+
+	return nil
+}
+
+// RestartNomadAlloc - restart any Nomad allocation of the associated job in NomadClient
+func (nc NomadClient) RestartNomadAlloc() error {
+	allocs, _, err := nc.client.nomad.Jobs().Allocations(nc.JobName, true, nil)
+	if err != nil {
+		return err
+	}
+	if len(allocs) > 0 {
+		allocID := ""
+		for i := range allocs {
+			if allocs[i].ClientStatus == "running" {
+				allocID = allocs[i].ID
+				break
+			}
+		}
+		if allocID == "" {
+			return fmt.Errorf("no running allocations to restart")
+		}
+		alloc, _, err := nc.client.nomad.Allocations().Info(allocID, nil)
+		if err != nil {
+			return err
+		}
+		logging.Info("[restart log] stopping %s", alloc.ID)
+		_, err = nc.client.nomad.Allocations().Stop(alloc, nil)
```

Comment:
The last command performed here is `Stop`. Are you relying on the Nomad replica count to recover the lost count -> "restart"?

Reply:
Yes

Reply:
Well the naming is a bit of misnomer, but I am fine with it la.

My suggestion then is to just add comments at the `Stop` part to state about the Nomad server recovering lost count as "restart"?

Reply:
Added the comment in 926bb901308c9af0e87a694a4a4dc4a499fff061